contrib/annoy-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+set(ANNOY_PROJECT_DIR "${ClickHouse_SOURCE_DIR}/contrib/annoy")
+set(ANNOY_SOURCE_DIR "${ANNOY_PROJECT_DIR}/src")
+
+add_library(_annoy ${ANNOY_SOURCE_DIR}/annoylib.h)
+target_include_directories(_annoy SYSTEM PUBLIC ${ANNOY_SOURCE_DIR})
+set_target_properties(_annoy PROPERTIES LINKER_LANGUAGE CXX)
```

Comment:
Without this line this will happen (some versions before) https://s3.amazonaws.com/clickhouse-builds/37215/bdb66914051da3376fd2d65ecc93cad757ab22ce/binary_splitted/build_log.log
CMake Error: CMake can not determine linker language for target: _spotify_annoy